Example_Programs/programming_concpets/tools_examples/gdb-examples/example3.c

```diff
@@ -0,0 +1,21 @@
+#include<stdio.h>
+
+int val;
+int *fetch() {
+  val=100;
+  return &val; 
+}
+
+int main(int argc, char* argv[]) {
+  int *ptr;
+  ptr=fetch(); 
+  printf("Ret val=%d\n",*ptr); 
+  return 0;                    
+}
+/*
+  In original code, val had local scope and hence after the 
```

Comment:
In simple words its called dangling pointer. Referring to address which is no long in scope.